src/renderers/dom/shared/__tests__/ReactDOMAttribute-test.js

```diff
@@ -133,5 +133,20 @@ describe('ReactDOM unknown attribute', () => {
       );
       expectDev(console.error.calls.count()).toBe(1);
     });
+
+    it('removes camelCased unknown props and warns', () => {
+      spyOn(console, 'error');
+
+      var el = document.createElement('div');
+      ReactDOM.render(<div hELLo="something" />, el);
+      expect(el.firstChild.hasAttribute('hELLo')).toBe(false);
```

Comment:
Even if we add this warning I don't think we plan to strip them. Rather, we'll warn but leave them in the DOM.